reconcile/gitlab_owners.py

```diff
@@ -187,33 +191,82 @@ def format_report(report):
         """
         markdown_report = ''
 
-        approvers = list()
+        closest_approvers = list()
         for _, owners in report.items():
             new_group = list()
 
-            if 'approvers' not in owners:
+            if 'closest_approvers' not in owners:
                 continue
 
-            for owner in owners['approvers']:
+            for closest_approver in owners['closest_approvers']:
                 there = False
 
-                for group in approvers:
-                    if owner in group:
+                for group in closest_approvers:
+                    if closest_approver in group:
                         there = True
 
                 if not there:
-                    new_group.append(owner)
+                    new_group.append(closest_approver)
 
             if new_group:
-                approvers.append(new_group)
+                closest_approvers.append(new_group)
 
-        if approvers:
-            markdown_report += (f'{COMMENT_PREFIX} You will need a "/lgtm" '
-                                f'from one person from each of these '
-                                f'groups:\n\n')
-        for group in sorted(approvers):
+        if closest_approvers:
+
+            if len(closest_approvers) == 1:
+                markdown_report += (f'{COMMENT_PREFIX} You will need a '
+                                    f'"/lgtm" from at least one person from '
+                                    f'the following group:\n\n')
+            else:
+                markdown_report += (f'{COMMENT_PREFIX} You will need a '
```

Comment:
when this can happen? Shouldn't the closest approvers group be unique?

Reply:
It's unique per change. This will happen when you change two files on different directories. Each file can then have its own set of closest approvers. Example:
https://gitlab.cee.redhat.com/service/managed-tenants/-/merge_requests/471